core/commands/storage.go

```diff
@@ -355,8 +510,8 @@ the chunk and replies back to client for the next challenge step.`,
 			Id: chidInt64,
 		}
 		chunkInfo := ss.GetOrDefault(chunkHash)
-		chunkInfo.NewChunk(pid, n.Identity, chID, price)
-		chunkInfo.SetState(initState)
+		chunkInfo.UpdateChunk(pid, n.Identity, chID, price)
+		chunkInfo.SetState(storage.StdChunkStateFlow[initState].State)
```

Comment:
Since `storage.StdChunkStateFlow` is a global var, you can just pass `chunkInfo.SetState(initState)` and get it from `storage.StdChunkStateFlow` inside set state

Reply:
I change const initState to be an int, since we need the numeric order for those steps. Are you suggesting also to use int as state identifier in chunkInfo? I would say they are both same effort, either way we need to convert int to string.

Reply:
I mean, the `storage.StdChunkStateFlow[state].State` can be done inside `chunkInfo.SetState` function, so you don't need to write out this map every time.